pipeline/scripts/13-excelExport.R

```diff
@@ -264,143 +267,217 @@ IS_neg$Intensity <- as.numeric(as.character(IS_neg$Intensity))
+IS_summed$Sample_level <- factor(IS_summed$Sample, levels = c(Sample_naturalorder))
+IS_pos$Sample_level <- factor(IS_pos$Sample, levels = c(Sample_naturalorder))
+IS_neg$Sample_level <- factor(IS_neg$Sample, levels = c(Sample_naturalorder))
+
+##########
+##### bar plots with all IS
+##########
+
+# function for ggplot theme
+# theme for all IS bar plots
+theme_IS_bar <- function(myPlot) {
+  myPlot + 
+    scale_y_continuous(breaks = scales::pretty_breaks(n = 10)) +
+    theme(
+      legend.position='none',
+      axis.text.x=element_text(angle = 90, hjust = 1, vjust = 0.5, size=6),
+      axis.text.y=element_text(size=6))
+}
 
-
-# Barplot for all IS (interne standaards)
-IS_neg_plot <- ggplot(IS_neg, aes(Sample,Intensity)) +
+# ggplot functions
+IS_neg_bar_plot <- ggplot(IS_neg, aes(Sample_level, Intensity)) +
   ggtitle("Interne Standaard (Neg)") +
   geom_bar(aes(fill=HMDB.name),stat='identity') +
   labs(x='',y='Intensity')+
-  facet_wrap(~HMDB.name, scales='free_y') +
-  theme(axis.text.x=element_text(angle = 90, hjust = 1, vjust = 0.5, size=8), legend.position='none') +
-  scale_y_continuous(breaks = scales::pretty_breaks(n = 10))
+  facet_wrap(~HMDB.name, scales='free_y')
 
-IS_pos_plot <- ggplot(IS_pos, aes(Sample,Intensity)) +
+IS_pos_bar_plot <- ggplot(IS_pos, aes(Sample_level, Intensity)) +
   ggtitle("Interne Standaard (Pos)") +
   geom_bar(aes(fill=HMDB.name),stat='identity') +
   labs(x='',y='Intensity') +
-  facet_wrap(~HMDB.name, scales='free_y') +
-  theme(axis.text.x=element_text(angle = 90, hjust = 1, vjust = 0.5, size=8), legend.position='none') +
-  scale_y_continuous(breaks = scales::pretty_breaks(n = 10))
+  facet_wrap(~HMDB.name, scales='free_y')
 
-IS_sum_plot <- ggplot(IS_summed, aes(Sample,Intensity)) +
+IS_sum_bar_plot <- ggplot(IS_summed, aes(Sample_level, Intensity)) +
   ggtitle("Interne Standaard (Summed)") +
   geom_bar(aes(fill=HMDB.name),stat='identity') +
   labs(x='',y='Intensity') +
-  facet_wrap(~HMDB.name, scales='free_y') +
-  theme(axis.text.x=element_text(angle = 90, hjust = 1, vjust = 0.5, size=8), legend.position='none') +
-  scale_y_continuous(breaks = scales::pretty_breaks(n = 10))
-
-
-len <- length(repl.pattern)
-
-w <- 9 + 0.35 * len
-ggsave(paste0(outdir, "/plots/IS_bar_all_neg.png"), plot=IS_neg_plot, height=w/2.5, width=w, units="in")
-ggsave(paste0(outdir, "/plots/IS_bar_all_pos.png"), plot=IS_pos_plot, height=w/2.5, width=w, units="in")
-ggsave(paste0(outdir, "/plots/IS_bar_all_sum.png"), plot=IS_sum_plot, height=w/2.5, width=w, units="in")
-
+  facet_wrap(~HMDB.name, scales='free_y')
+
+# add theme to ggplot functions
+IS_neg_bar_plot <- theme_IS_bar(IS_neg_bar_plot)
+IS_pos_bar_plot <- theme_IS_bar(IS_pos_bar_plot)
+IS_sum_bar_plot <- theme_IS_bar(IS_sum_bar_plot)
+
+# save plots to disk
+w <- 9 + 0.35 * sample_count
+ggsave(paste0(outdir, "/plots/IS_bar_all_neg.png"), plot=IS_neg_bar_plot, height=w/2.5, width=w, units="in")
```

Comment:
height_plot variable at line 321 instread of three times w/2.5?

Reply:
Optional to make ggsave a function? Seems a bit repetitive now.

Reply:
agreed. I want to give this script and all other R-scripts an overhaul in DIMS3.0. Many scripts have similar inconsistencies or outdated code/comments. Additionally, because of the module load instead of containers, workarounds were build that created bigger code. I rather want to tackle this comment then. I keep this in mind.